systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -86,21 +86,26 @@ public static void waitForTopicsReady(String namespaceName, String topicPrefix,
         waitForTopicStatus(namespaceName, topicPrefix, numberOfTopics, CustomResourceStatus.Ready);
     }
 
-    public static void waitForTopicsContainConfig(String namespaceName, String topicPrefix, int numberOfTopics, Map<String, Object> config) {
-        LOGGER.info("Verifying that {} Topics contain right config", numberOfTopics);
-        List<CompletableFuture<?>> topics = new ArrayList<>();
+    /**
+     * Waits for a specific range of Kafka topics to contain the specified configurations.
+     *
+     * @param namespaceName The namespace in which the topics reside.
+     * @param topicPrefix The common prefix of the topic names.
+     * @param startIndex The starting index of topics to check.
+     * @param endIndex The ending index of topics to check (exclusive).
+     * @param config The configurations to verify within the topics.
+     */
+    public static void waitForTopicsContainConfig(String namespaceName, String topicPrefix, int startIndex, int endIndex, Map<String, Object> config) {
+        LOGGER.info("Verifying that Topics from index {} to {} contain the correct config", startIndex, endIndex - 1);
 
-        for (int i = 0; i < numberOfTopics; i++) {
-            String currentTopic = topicPrefix + i;
-            topics.add(CompletableFuture.runAsync(() -> {
-                KafkaTopicUtils.waitForTopicConfigContains(namespaceName, currentTopic, config);
-            }));
+        for (int i = startIndex; i < endIndex; i++) {
+            String currentTopic = topicPrefix + "-" + i;
+            KafkaTopicUtils.waitForTopicConfigContains(namespaceName, currentTopic, config);
```

Comment:
It was not so much faster as you would expect as I have tested it...There were some hidden a bug when something went wrong resulting in prolonging execution time (one would expect that when one WaitException arises) it would mean that everything will stop but in that case other threads don't know about that so they continue to spawning more topics....until KafkaTopic will not be ready. So I have switched to sequential here because of better debugging, plus such methods are then used in concurrent execution in mine new created methods.

Reply:
Okay, makes sense, thanks.